src/components/MultiHopTrade/hooks/quoteValidation/usePriceImpact.tsx

```diff
@@ -68,5 +68,8 @@ export const usePriceImpact = (tradeQuote: TradeQuote | undefined) => {
     return priceImpactPercentage.gt(10)
   }, [priceImpactPercentage])
 
-  return { isModeratePriceImpact, priceImpactPercentage, isHighPriceImpact }
+  return useMemo(
```

Comment:
actually it does, since the object wrapping those memoized values is created each time the function is invoked which creates a new reference and triggers a rerender in the consumer

Reply:
But since this is done at the very last line of the body of the hook, the line of thinking here is this would be a no-op if nothing in the hook warranted a re-render, regardless of whether the return value itself is memoized. See this guy above https://github.com/shapeshift/web/pull/6338#discussion_r1544625340

Now parents consuming these are a different story - if they re-render and call this guy again, indeed that would be useful to avoid re-renders!

Happy to keep the `useMemo` (although not as an explicit return but first being assigned to a constant before being returned) for paranoia as I'm very much in for "memoize all the things".

Reply:
done ser
5e35960fcdd69296fff2972739c36bdf9af68c4c